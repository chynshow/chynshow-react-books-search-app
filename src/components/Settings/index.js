import React, { useContext, useState } from 'react';
import Input from '../Input';
import { AppContext } from '../../state/AppContext';
import { AlertContext } from '../../state/AlertContext';

const Settings = () => {
  // Get data from global app state
  const { showAlert } = useContext(AlertContext);
  const { maxResults, setMaxResults } = useContext(AppContext);
  // Use local state for form handling
  const [value, setValue] = useState(maxResults);
  // Handle onSubmit event
  const handlerOnSubmit = (e) => {
    e.preventDefault();
    setMaxResults(value);
    showAlert('Applied new search parameter value!');
  };
  return (
    <form onSubmit={handlerOnSubmit}>
      <Input
        className='c-setting-form__input'
        name='maxResults'
        type='number'
        value={value}
        placeholder='Max Results'
        onChange={(e) => setValue(e.target.value)}
      />
      <button className='c-btn c-settings-form__btn'>Apply</button>
    </form>
  );
};

export default Settings;
