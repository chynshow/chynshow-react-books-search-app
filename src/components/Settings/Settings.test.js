import React from 'react';
import Settings from './';
import { shallow } from 'enzyme';

it('should renders Settings component', () => {
  const c = shallow(<Settings />);
  expect(c).toMatchSnapshot();
});
