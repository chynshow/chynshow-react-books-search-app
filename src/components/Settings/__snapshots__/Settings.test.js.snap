// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should renders Settings component 1`] = `
<div
  className="c-settings"
>
  <h3
    className="c-settings__title"
  >
    Settings
  </h3>
  <form
    className="c-settings-form"
    onSubmit={[Function]}
  >
    <Memo(Input)
      className="c-settings-form__input"
      label="Max Results"
      name="maxResults"
      onChange={[Function]}
      placeholder="Max Results"
      type="number"
      value={10}
    />
    <div
      className="c-settings-form__btn-box"
    >
      <button
        className="c-btn c-btn--primary c-settings-form__btn"
      >
        Apply
      </button>
      <button
        className="c-btn c-btn--secondary c-settings-form__btn"
        onClick={[Function]}
      >
        Cancel
      </button>
    </div>
  </form>
</div>
`;
